components/org.wso2.carbon.identity.media.endpoint/src/main/java/org/wso2/carbon/identity/media/endpoint/service/MediaService.java

```diff
@@ -307,11 +250,69 @@ private DataContent downloadMedia(String type, String id) {
      */
     private void validateAllowedMediaTypes(String mediaTypePathParam) {
 
-        List<String> allowedContentTypes = loadAllowedContentTypes();
-        if (CollectionUtils.isEmpty(allowedContentTypes) || !allowedContentTypes.contains(mediaTypePathParam)) {
+        StorageSystemManager storageSystemManager = getStorageSystemManager();
+        try {
+            storageSystemManager.validateMediaTypePathParam(mediaTypePathParam);
+        } catch (StorageSystemException e) {
             throw handleException(Response.Status.FORBIDDEN,
                     MediaServiceConstants.ErrorMessage.ERROR_CODE_ERROR_UNSUPPORTED_CONTENT_TYPE_PATH_PARAM);
         }
     }
 
+    /**
+     * Delete a media.
+     *
+     * @param type The high level content-type of the resource (if media content-type is image/png then
+     *             type would be image.
+     * @param id   Unique identifier for the requested media file to be deleted.
+     */
+    public void deleteMedia(String type, String id) {
+
+        validateAllowedMediaTypes(type);
+        StorageSystemManager storageSystemManager = getStorageSystemManager();
+        String tenantDomain = getTenantDomainFromContext();
+        try {
+            boolean mediaDeleted = storageSystemManager.isMediaDeleted(id, type, tenantDomain);
```

Comment:
Is this a media existence check? If so we normally follow `isMediaExist` format.

Reply:
this method would delete the media and return true if the delete has been successful

Reply:
The name will give an incorrect meaning IMO. Shall we change the delete method as void and throw an exception if something goes wrong during the deletion process. Then catch that exception and handle it from here accordingly.

Reply:
fixed with: 0d72742b085d7a3aa6c70ce0860be0a1580b9715